### DISCUSSION TITLE
Playing radio in real time

### DISCUSSION THREAD

**Discussion by nottledim** [schreibfaul1/ESP32-audioI2S#657] [General]
This has me completely perplexed and somewhat amused:  I'm using the simple example program with a UM Tiny Pico esp32 board that includes psram and a PCM5102 to receive BBC Radio 3.  This is my first play with these toys so I'm at the bottom of the learing curve.

 It sort of works!  However it does not play in current time, it plays the radio from about 6hrs ago.

It downloads a playlist of about 6000 lines and then reports a media sequence number like 266456056. Starting from that number it then downloads a ts file and then incrementing from that number downloads the next file that at roughly 7s intervals.

After 5 mins or so it stops playing and downloads another 6000 line file and then carries on playing where it left off.

`17:48:50.007 -> info        new request: "http://as-hls-ww-live.akamaized.net/pool_904/live/ww/bbc_radio_three/bbc_radio_three.isml/bbc_radio_three-audio=320000.m3u8"
17:48:50.714 -> info        current playlist line: 1000
17:48:51.355 -> info        current playlist line: 2000
17:48:51.997 -> info        current playlist line: 3000
17:48:52.639 -> info        current playlist line: 4000
17:48:53.312 -> info        current playlist line: 5000
17:48:53.954 -> info        current playlist line: 6000
17:48:54.500 -> info        new request: "http://as-hls-ww-live.akamaized.net/pool_904/live/ww/bbc_radio_three/bbc_radio_three.isml/bbc_radio_three-audio=320000-266456151.ts"
17:49:00.792 -> info        new request: "http://as-hls-ww-live.akamaized.net/pool_904/live/ww/bbc_radio_three/bbc_radio_three.isml/bbc_radio_three-audio=320000-266456152.ts"
17:49:07.210 -> info        new request: "http://as-hls-ww-live.akamaized.net/pool_904/live/ww/bbc_radio_three/bbc_radio_three.isml/bbc_radio_three-audio=320000-266456153.ts"`

I'm curious to understand why it's behaving this way and what I can do correct it.

**Comment by schreibfaul1** [comment]
The URL returns a playlist that is over 200KB in size. A µC such as the ESP32 needs some time to process this.
This URL is more suitable: http://as-hls-ww-live.akamaized.net/pool_904/live/ww/bbc_radio_three/bbc_radio_three.isml/bbc_radio_three-audio=48000.norewind.m3u8 with ~600 bytes and 5 entries each.

**Comment by nottledim** [comment]
Oh yes, that works a treat.  I've got a lot to learn!

Many thanks
